### PR TITLE
Update to platform-tools-31.0.0

### PR DESCRIPTION
**Pull request by Johnnynator**
Adb tested runtime on x86_64

Build (with GCC 10) for
```
aarch64 w musl-1.1.24
aarch64 w glibc 2.32
armv6l w musl-1.1.24
armv6l w glibc 2.32
armv7l w glibc 2.32
armv7l w musl-1.1.24
i686 w glibc 2.32
x86_64 w musl-1.1.24
x86_64 w glibc 2.32
```

There didn't seem all to big changes, most of the lines are just due to changed file paths

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ set(ANDROID_MKE2FS_NAME "mke2fs.android")
 
 # Version of android-tools and the version of boringssl being used.
 # See: https://android.googlesource.com/platform/external/boringssl/+/platform-tools-${ANDROID_VERSION}/BORINGSSL_REVISION
-set(ANDROID_VERSION 30.0.5p1)
+set(ANDROID_VERSION 31.0.0)
 set(BORINGSSL_VERSION 2fb729d4f36beaf263ad85e24a790b571652679c)
```
Comment (nmeum) [inline CMakeLists.txt:15]: https://android.googlesource.com/platform/external/boringssl/+/platform-tools-31.0.0/BORINGSSL_REVISION

Seems to me we need to update this to `c1e156ae16fa8b61af9b5d2b74e59d3f86e141be`.

Comment (Johnnynator) [reply CMakeLists.txt:15]: Oh yeah, totally forgot about boringssl

File: CMakeLists.txt [hunk 2]
```diff
@@ -12,7 +12,7 @@ set(ANDROID_MKE2FS_NAME "mke2fs.android")
 # Version of android-tools and the version of boringssl being used.
 # See: https://android.googlesource.com/platform/external/boringssl/+/platform-tools-${ANDROID_VERSION}/BORINGSSL_REVISION
 set(ANDROID_VERSION 31.0.0)
-set(BORINGSSL_VERSION 2fb729d4f36beaf263ad85e24a790b571652679c)
+set(BORINGSSL_VERSION c1e156ae16fa8b61af9b5d2b74e59d3f86e141be)
```
Comment (anatol) [inline CMakeLists.txt:15]: it makes sense to squash this change into the original 31.0.0 version bump commit

File: CMakeLists.txt (hunk 1 above)
Comment (anatol) [reply CMakeLists.txt]: Is there any way to check at build-time (in CMake files) that the boringssl tree corresponds to file BORINGSSL_REVISION?

`rev=$(cd vendor/boringssl; git rev-list -n 1 HEAD)`

Such automatic check would eliminate such issues in the future.

Comment (nmeum) [reply CMakeLists.txt]: Probably is with [execute_process](https://cmake.org/cmake/help/latest/command/execute_process.html). Wouldn't do that as part of this PR though.

**Comment by nmeum** [comment]
Did some local tests on Alpine the last few days, worked fine and cherry picked this in 20407b6b18994a21c0c83fb4b5963280287df6a8 therefore. We can do some further tests on other setups before making a new release. Thanks for your work on this! :)

**Pull request closed without merging**